### ISSUE TITLE
[Android] Crash when Starting Core

### ISSUE THREAD

**Issue by MajorWilly** [libretro/libretro-uae#81]
Hi, I was really excited when I noticed that this core was being updated, especially when I saw the great work that @sonninnos and @rsn8887 had done with VICE. Unfortunately it crashes when Starting the core. I had an Android crashing problem on Vice (which was fixed), so maybe related ?

I'm using -

RetroArch Nightly v1.7.7 b4375e3
PUAE 2.6.1 4f66df9

Android Kitkat 4.4.4
armeabi-v7a
Mali-T764

[adb log.txt](https://github.com/libretro/libretro-uae/files/3540653/adb.log.txt)

**Comment by rsn8887** [comment]
I can reproduce this bug on my Nvidia Shield Portable.

Unfortunately, the fix that worked on Vice (setting optimizations to -O0 or -Ofast) doesn't help and I have no clue what causes this, or even if it is just Kitkat at the moment.

**Comment by rsn8887** [comment]
Has this core ever worked on Kitkat?

**Comment by MajorWilly** [comment]
It's good to hear that you also get the crash. I did try it about a year ago and it didn't work then either ! I hope it can be fixed as this is my number 1 retro system ! (Amiga, Spectrum, C64). 

Android UAE4Arm (by lubomyr) barely works on any Android device so this got me excited when I saw the news on the Libretro site.

**Comment by rsn8887** [comment]
@Sonninnos got it to boot up on his ARMv7 phone. But it shows "failed to load content" there, and no .adf can be loaded, always says "failed to load content".

**Comment by MajorWilly** [comment]
It said "failed to load content" to begin with for me as well, but it was because I forgot to put the kickstarts in the System directory ! Now it just crashes as you know.

**Comment by rsn8887** [comment]
Yeah I just realized that. I guess some serious debugging is needed. There are a bunch of ANDROID defines, but nothing major, and I couldn't see any suspicious compilation warnings. It would be good to know if this is limited to older Android versions like kitkat, or happens on ALL Android devices.

**Comment by MajorWilly** [comment]
There is another Android emulator called 'Bluestacks' which uses Android 7. I haven't tried it though. Not sure what version of Android is on MEmu.

**Comment by sonninnos** [comment]
I did get kickstart to show on MEmu, but crashed when opened RA menu.

**Comment by MajorWilly** [comment]
So is it having trouble with the Kickstarts from the System directory ? I ask this because the Atari ST core Hatari also doesn't work properly. It says the TOS bios is invalid even though it's the exact required checksum.

**Comment by MajorWilly** [comment]
Just out of interest, are there any possibilities you guys have in mind that could fix this, or you both have no idea and I should crush my hopes now ?! Just asking.

**Comment by sonninnos** [comment]
Since it does work on MEmu it is probably something simple, but I have no idea yet.

**Comment by gouchi** [comment]
I reproduce the issue using RA32 and older Android version 4.2.2.

We can try also to use [Android x86 Kit Kat](https://osdn.net/projects/android-x86/downloads/65695/android-x86-4.4-r5.iso/).

I will try to make some test with newer Android.

**Comment by gouchi** [comment]
I reproduce the issue using RA32 and older Android version 4.2.2.

We can try also to use [Android x86 Kit Kat](https://osdn.net/projects/android-x86/downloads/65695/android-x86-4.4-r5.iso/).

I will try to make some test with newer Android.

RA64 seems to have worked according to [forum post](https://forums.libretro.com/t/amiga-p-uae-core/6728/28).

**Comment by MajorWilly** [comment]
I haven't tried this yet as i'm somewhere else and I doubt the buildbot is ready yet anyway. If this is fixed then thankyou all. 

UAE4Arm is a great emulator (It still doesn't have Cycle Accuracy as far as i'm aware). Unfortunately it locks up when you use the virtual keyboard ! Only a very old version works that's why I had my hopes up for this core to become working.

I'll let you know if it works a bit later.

**Comment by MajorWilly** [comment]
@rsn8887 I was wondering why the buildbot was taking so long ! Oh well.

A quick question before I start using it tomorrow (if it works !). Is the CAPS library compatible with the Android version and if so what file version to use (.so ?). Does it still have to be placed in the RetroArch directory like the Windows version ? I thought it would be more logical to have it in the System directory.

**Comment by MajorWilly** [comment]
Any way you could change it to the System directory so we don't require ROOT access ? If not i'll just have to use ADFs.

**Comment by sonninnos** [comment]
I changed the CAPS directory to system directory in my latest PR.

**Comment by MajorWilly** [comment]
@rsn8887 Thanks, that fix works. I'm quite surprised at how good the compatibility is with this core ! I get a little bit of slowdown with games that use Copper. It's a shame this emu doesn't have the 'Fast Copper' hack that the others do. Overall though, it's a great core.

@sonninnos Unfortunately, I couldn't get any IPF images to load on Android. I used the Linux 32bit 'capsimg.so' in the System directory. Is there a specific one for Android or which one would be Android compatible ?
